src/test/kotlin/com/bardsoftware/papeeria/backend/cosmas/CosmasGoogleCloudServiceTest.kt

```diff
@@ -200,6 +201,59 @@ class CosmasGoogleCloudServiceTest {
         assertNotNull(stream.error)
     }
 
+    @Test
+    fun simpleAddPatch() {
+        createVersion("file", "1", "1")
+        addPatchToService("abc", "-", "1", 1, "1")
+        commit("1")
+    }
+
+    @Test
+    fun checkEmptyList() {
+        createVersion("file", "1", "1")
+        addPatchToService("abc", "-", "1", 1, "1")
+        commit("1")
+        createVersion("file1", "1", "1")
+        val list = this.service.getPatchList("1", "1")
+        assertEquals(0, list?.size)
+    }
+
+    @Test
+    fun getPatchSimple() {
+        createVersion("file", "1", "1")
+        addPatchToService("abc", "-", "1", 1, "1")
+        addPatchToService("abc1", "-1", "1", 12, "1")
+        commit("1")
+        val list = this.service.getPatchListFromMemory("1", 0)
+        assertEquals("abc", list?.get(0)?.text)
+        assertEquals("-", list?.get(0)?.userId)
+        assertEquals("abc1", list?.get(1)?.text)
+        assertEquals("-1", list?.get(1)?.userId)
+        assertEquals(1L, list?.get(0)?.timeStamp)
+        assertEquals(12L, list?.get(1)?.timeStamp)
+    }
+
+    @Test
+    fun getPatchManyVersions() {
+        createVersion("file", "1", "1")
+        addPatchToService("abc", "-", "1", 1, "1")
+        addPatchToService("abc1", "-1", "1", 12, "1")
+        commit("1")
+        createVersion("file2", "1", "1")
+        addPatchToService("abcd", "-d", "1", 13, "1")
+        addPatchToService("abcd1", "-d1", "1", 14, "1")
+        commit("1")
+        assertEquals(getFileFromService(0, "1", "1"), getFileFromService(1, "1", "1"))
+        val list = this.service.getPatchListFromMemory("1", 1)
```

Comment:
any sense to check that patch list from the previous version is also available?

Reply:
ok